#! /bin/sh /usr/share/dpatch/dpatch-run
## 10_fix_ati_crash.dpatch by  Bradley Smith <brad@brad-smith.co.uk>
##
## DP: Fixes crash on ati cards

@DPATCH@

diff -Naur briquolo-0.5.7.orig/src/MOGL/MOGL_Fenetre.cpp briquolo-0.5.7/src/MOGL/MOGL_Fenetre.cpp
--- briquolo-0.5.7.orig/src/MOGL/MOGL_Fenetre.cpp	2008-03-24 13:23:07.000000000 +0000
+++ briquolo-0.5.7/src/MOGL/MOGL_Fenetre.cpp	2008-03-29 19:41:40.000000000 +0000
@@ -168,6 +168,11 @@
 			// On recommence donc !
 			_NbBitParPixel = 24;
 			_SurfaceSDL = _CreerFenetre(_NbBitParPixel, flags);
+           if(_SurfaceSDL == NULL)
+           {
+               _NbBitParPixel = 16;
+               _SurfaceSDL = _CreerFenetre(_NbBitParPixel, flags);
+           }
 		}
 	}
 	if (_SurfaceSDL == NULL)
